src/IK/src/GravityTask.cpp

```diff
@@ -163,46 +147,35 @@ bool GravityTask::update()
 
     m_isValid = false;
 
-    // Normalize Accelerometer vector
-    m_accDenomNorm = sqrt(pow(m_currentAcc(0),2) + pow(m_currentAcc(1),2) + pow(m_currentAcc(2),2));
-
-    for (size_t i = 0; i < m_currentAcc.size(); i++)
-    {
-        m_currentAccNorm(i) = m_currentAcc(i) / m_accDenomNorm;
-    }
+    auto targetRotation = toEigen(m_kinDyn->getWorldTransform(m_targetFrameIndex).getRotation());
 
-    m_currentAccNorm = toEigen(m_kinDyn->getWorldTransform(m_targetFrameName).getRotation()) * m_currentAccNorm;
+    Eigen::Vector3d desiredZDirectionAbsolute = targetRotation * m_desiredZDirectionBody;
+    Eigen::Vector3d feedforwardAbsolute = targetRotation * m_feedForwardBody;
 
-    // get the relative jacobian
-    if (!m_kinDyn->getRelativeJacobian(m_baseIndex, m_targetFrameIndex, m_relativeJacobian))
+    if (!m_kinDyn->getFrameFreeFloatingJacobian(m_targetFrameIndex, m_jacobian))
```

Comment:
What do you think of this change? 

```suggestion
    if ( m_baseName == "")
    {
        if (!m_kinDyn->getFrameFreeFloatingJacobian(m_targetFrameIndex, m_jacobian))
    {
        log()->error("[GravityTask::update] Unable to get the frame jacobian.");
        return m_isValid;
    }
    }
    else
        if (!m_kinDyn->getRelativeJacobian(m_baseIndex, m_targetFrameIndex, m_jacobian))
    {
        log()->error("[GravityTask::update] Unable to get the relative jacobian.");
        return m_isValid;
    }
```

Reply:
It does not work, since also the rotation matrix need to change accordingly. Moreover, the dimensions are not correct